pymilvus/milvus_client/milvus_client.py

```diff
@@ -0,0 +1,801 @@
+from pymilvus.orm.connections import connections
+from pymilvus.orm.types import DataType, infer_dtype_bydata
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+
+
+class MilvusClient:
+    """The Milvus Client"""
+    def __init__(
+        self,
+        collection_name: str = "ClientCollection",
+        pk_field: str = None,
+        vector_field: str = None,
+        uri: str = None,
+        shard_num: int = 2,
+        partitions: List[str] = None,
+        consistency_level: str = "Session",
+        replica_number: int = 1,
+        index_params: dict = None,
+        timeout: Optional[int] = None,
+        drop_old: bool = False,
+    ):
+        """A client for the common Milvus use case.
+
+        This client attempts to hide away the complexity of using Pymilvus. In a lot ofcases what
+        the user wants is a simple wrapper that supports adding data, deleting data, and searching.
+        This wrapper can autoinfer the schema from a previous collection or newly inserted data,
+        can update the paritions, can query, and can delete by pk.
+
+        Args:
+            pk_field (str, optional): Which entry in data is considered the primary key. If None,
+                an auto-id will be created. Will be overwritten if loading from a previous
+                collection. Defaults to None.
+            vector_field (str, optional): Which entry in the data is considered the vector field.
+                Will get overwritten if loading from previous collection. Defaults to None.
+            uri (str, optional): The connection address to use to connect to the
+                instance. Defaults to "http://localhost:19530". You can also set this address
+                as an env variable  
+            shard_num (int, optional): The amount of shards to use for the collection. Unless
+                dealing with huge scale, recommended to keep at default. Defaults to 2.
+            partitions (List[str], optional): Which paritions to create for the collection.
+                Defaults to None.
+            consistency_level (str, optional): Which consistency level to use for the Client.
+                The options are "Strong", "Bounded", "Eventually", "Session". Defaults to "Session".
+            replica_number (int, optional): The amount of in memomory replicas to use.
+                Defaults to 1.
+            index_params (dict, optional): What index parameteres to use for the Collection.
+                If none, will use a default one. Defaults to None.
+            timeout (Optional[int], optional): What timeout to use for function calls. Defaults
+                to None.
+            drop_old (bool, optional): If a collection with the same name already exists, drop it.
+                Defaults to False.
+        """
+        self.uri = uri
+        self.collection_name = collection_name
+        self.shard_num = shard_num
+        self.partitions = partitions
```

Comment:
we should just ignore the partition here.

Later on user can specify the partition key here

Reply:
Milvus will automatially create 64 or 256 partition for the user

Reply:
They still need to name the partitions correct? Should I instead do the partition check on each insert?